### PR TITLE
Fix some crashes related to realm #4454

### LINKED ISSUE #4454: Fix some crashes related to realm

### PR DESCRIPTION
**Pull request by oa-s**
Closes #4454

### CODE REVIEW

**Comment by oa-s** [comment]
According to comments from:

https://github.com/realm/realm-core/issues/5043
https://github.com/realm/realm-swift/issues/5433

File: AlphaWallet/Core/Types/RealmStore.swift [hunk 1]
```diff
@@ -24,9 +25,9 @@ final class RealmStore {
             callback(realm)
         } else {
             dispatchPrecondition(condition: .notOnQueue(queue))
-            queue.sync { [unowned self] in
+            queue.sync {
                 autoreleasepool {
-                    let realm = try! Realm(configuration: config)
```
Comment (hboon) [inline AlphaWallet/Core/Types/RealmStore.swift:29]: Why does the `try! Realm(configuration:)` call fail sometimes?

Comment (oa-s) [reply AlphaWallet/Core/Types/RealmStore.swift:29]: its not, just to be sure

File: AlphaWallet/Core/Types/RealmStore.swift [hunk 2]
```diff
@@ -43,3 +44,63 @@ extension Realm {
...
     }
 }
+
+protocol DetachableObject: AnyObject {
+    func detached() -> Self
+}
+
+extension Object: DetachableObject {
+
+    func detached() -> Self {
```
[the remark this answers is not in the document]
Comment (oa-s) [reply AlphaWallet/Core/Types/RealmStore.swift:54]: it makes copy of object that can be passed later, .frozen() doesn't. looks ok, for long time storing out of comments from links above.
i was facing few crashes related to realm, these changes should fix an issue

[the remark this answers is not in the document]
Comment (hboon) [reply AlphaWallet/Core/Types/RealmStore.swift:54]: Oh, so this actually does what we expect `frozen()` should do.

**Review by hboon — APPROVED** [review]

File: AlphaWallet/Core/Types/RealmStore.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (oa-s) [reply AlphaWallet/Core/Types/RealmStore.swift:54]: Yep

**Pull request merged**